dash.el

```diff
@@ -1802,6 +1802,20 @@ kv can be any key-value store, such as plist, alist or hash-table."
      (t
       (cons (list s source) (dash--match-kv-1 (cdr match-form) s (car match-form)))))))
 
+(defun dash-expand:&hash (key source)
+  "Generate extracting KEY from SOURCE for &hash destructuring."
+  `(gethash ,key ,source))
+
+(defun dash-expand:&plist (key source)
+  "Generate extracting KEY from SOURCE for &plist destructuring."
+  `(plist-get ,source ,key))
+
+(defun dash-expand:&alist (key source)
```

Comment:
@Fuco1 	Add another destructuring key for structs

Reply:
Can you elaborate?

Reply:
Oh, these were notes for myself!

I was thinking we could write a `&struct` to be a generic one and you could also write your own like you did in an example above for lsp.  But I think we should keep that out of this PR as it's not really related.